sdk/src/main/scala/com/horizen/api/http/SidechainWalletApiRoute.scala

```diff
@@ -8,26 +8,30 @@ import com.horizen.SidechainNodeViewHolder.ReceivableMessages
 import com.horizen.SidechainNodeViewHolder.ReceivableMessages.LocallyGeneratedSecret
 import com.horizen.SidechainTypes
 import com.horizen.api.http.JacksonSupport._
-import com.horizen.api.http.SidechainWalletErrorResponse.ErrorSecretNotAdded
-import com.horizen.api.http.SidechainWalletRestScheme._
+import com.horizen.api.http.SidechainWalletErrorResponse.{ErrorFailedToParseSecret, ErrorPropositionNotFound, ErrorPropositionNotMatch, ErrorSecretAlreadyPresent, ErrorSecretNotAdded}
+import com.horizen.api.http.SidechainWalletRestScheme.{ReqDumpWallet, ReqExportSecret, ReqImportSecret, RespDumpSecrets, RespExportSecret, RespImportSecrets, _}
 import com.horizen.box.Box
-import com.horizen.proposition.{Proposition, VrfPublicKey}
+import com.horizen.companion.SidechainSecretsCompanion
+import com.horizen.proposition.{Proposition, PublicKey25519PropositionSerializer, VrfPublicKey}
 import com.horizen.secret.{PrivateKey25519Creator, VrfKeyGenerator}
 import com.horizen.serialization.Views
 import com.horizen.utils.BytesUtils
 import scorex.core.settings.RESTApiSettings
 
+import java.io.{File, PrintWriter}
 import scala.collection.JavaConverters._
 import scala.concurrent.{Await, ExecutionContext}
 import scala.util.{Failure, Success, Try}
-import java.util.{Optional => JOptional}
+import java.util.{Scanner, Optional => JOptional}
+import java.util.{ArrayList => JArrayList}
 
 case class SidechainWalletApiRoute(override val settings: RESTApiSettings,
-                                   sidechainNodeViewHolderRef: ActorRef)(implicit val context: ActorRefFactory, override val ec: ExecutionContext)
+                                   sidechainNodeViewHolderRef: ActorRef,
+                                   sidechainSecretsCompanion: SidechainSecretsCompanion)(implicit val context: ActorRefFactory, override val ec: ExecutionContext)
   extends SidechainApiRoute {
 
   override val route: Route = pathPrefix("wallet") {
-    allBoxes ~ coinsBalance ~ balanceOfType ~ createPrivateKey25519 ~ createVrfSecret ~ allPublicKeys
+    allBoxes ~ coinsBalance ~ balanceOfType ~ createPrivateKey25519 ~ createVrfSecret ~ allPublicKeys ~ importSecret  ~ exportSecret ~ dumpSecrets ~ importSecrets
```

Comment:
add authorization to `createPrivateKey25519` and `createVrfSecret`  methods.
The same for `transaction` api where we spend coins or send raw tx: `createCoreTransaction ~ createCoreTransactionSimplified ~  sendCoinsToAddress ~ sendTransaction ~ withdrawCoins ~ makeForgerStake ~ spendForgingStake`

Reply:
I tought this could be done as a separate task, but it's fine for me to do it now.

Reply:
right, let's do it in a separate PR. Also, we need to have a separate python or unit test that verifies authentication flow.